src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -194,46 +194,96 @@ && localCheckoutProvided()
         boolean failed = false;
         LOGGER.log(Level.INFO, "Starting plugin tests on core coordinates {0}", coreCoordinates);
         for (UpdateSite.Plugin plugin : data.plugins.values()) {
-            if (config.getIncludePlugins() == null
-                    || config.getIncludePlugins().contains(plugin.name.toLowerCase())) {
-                PluginInfos pluginInfos = new PluginInfos(plugin.name, plugin.version, plugin.url);
```

Comment:
Moved this down to where it is first used to reduce unnecessary scope and cognitive load.